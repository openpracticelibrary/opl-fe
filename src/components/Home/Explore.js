import React from 'react';
import {
  Stack,
  Box,
  Heading,
  Flex,
  Image,
  Button,
} from "@chakra-ui/core";
import { navigate } from "@reach/router";

import GrayRain from "../../assets/icons/rain.svg";
import ClosingStickies from "../../assets/images/home-practices.png";

const backgroundImage = `linear-gradient(180deg, #FFFFFF 0%, rgba(255,255,255,0.95) 71.03%, rgba(255,255,255,0.85) 100%), url(${GrayRain})`;

const Explore = (props) => (
  <Flex justify="center" minH={500} px={20} mt={10} bgImage={backgroundImage}>
    <Flex align="center" m={4}>
      <Image src={ClosingStickies} alt="Closing Message" width={561} height={357} />
      <Stack spacing={6} maxW={700} px={10}>
        <Heading fontSize="3xl" fontWeight="500" data-testid="exploreMsg">
          {props.exploreMessage}
        </Heading>
        <Box>
          <Button
            variantColor="lightBlue"
            size="lg"
            border="1px"
            borderColor="blue.500"
            rounded="32px"
            px={8}
            fontSize="md"
            color="black"
            fontFamily="heading"
            fontWeight="500"
            onClick={() => navigate("/practice")}
          >
             Explore the library
          </Button>
        </Box>
      </Stack>
    </Flex>
  </Flex>
);

export default Explore;
