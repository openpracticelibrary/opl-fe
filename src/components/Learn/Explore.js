import React from "react";
import { Button, Box, Typography } from "@material-ui/core";
import { navigate } from "@reach/router";

const Explore = props => {
  return (
    <Box
      display="flex"
      alignItems="center"
      alignContent="center"
      justifyContent="space-between"
      height="10rem"
      bgcolor="grey.200"
      px={4}
    >
      <Typography variant="h5" align="left" data-testid="exploreMsg">
        Learn something new from your peers and experiment with best practices!
      </Typography>
      <Button
        variant="outlined"
        color="primary"
        size="large"
        data-testid="exploreLibraryButton"
        onClick={() => navigate("/")}
      >
        Explore the Library
      </Button>
    </Box>
  );
};

export default Explore;
