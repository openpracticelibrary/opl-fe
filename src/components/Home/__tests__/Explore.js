import React from "react";
import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MockedProvider } from "@apollo/react-testing";
import renderer from 'react-test-renderer';
import Explore from '../Explore';

afterEach(cleanup);

it("renders the Explore hero without crashing", () => {
  const { getByTestId } = render(
    <MockedProvider mocks={[]}>
      <Explore exploreMessage="Wazzup" />
    </MockedProvider>
  );

  expect(getByTestId("exploreMsg")).toHaveTextContent("Wazzup");
});

it("matches the snapshot", () => {
  const tree = renderer.create(
    <MockedProvider mocks={[]}>
      <Explore exploreMessage="Snapshottin it up" />
    </MockedProvider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
