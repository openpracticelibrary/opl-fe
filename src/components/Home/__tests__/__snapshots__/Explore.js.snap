// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-84 makeStyles-explore-82"
  >
    <div
      class="MuiBox-root MuiBox-root-85"
    >
      <img
        alt="Closing Message"
        height="auto"
        src="closingstickies.png"
        width="60%"
      />
    </div>
    <div
      class="MuiBox-root MuiBox-root-86"
    >
      <div
        class="MuiBox-root MuiBox-root-87"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignLeft"
          data-testid="exploreMsg"
        >
          Snapshottin it up
        </h5>
        <div
          class="MuiBox-root MuiBox-root-118"
        >
          <button
            bg="#dff1ff"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained OplButton-root-119 OplButton-root-120 MuiButton-disableElevation"
            data-testid="exploreLibraryButton"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Explore the library
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
